src/core/Tree/TreeNode.tsx

```diff
@@ -0,0 +1,192 @@
+  /**
+   * Is TreeNode disabled.
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * Is TreeNode expanded.
+   * @default false
+   */
+  isExpanded?: boolean;
+  /**
+   * Is TreeNode selected.
+   * @default false
+   */
+  isSelected?: boolean;
+  /**
+   * Callback fired when expanding or closing a TreeNode.
+   */
+  onNodeExpanded?: (nodeId: string, expanded: boolean) => void;
+  /**
+   * Callback fired when selecting a TreeNode.
+   */
+  onNodeSelected?: (nodeId: string, selected: boolean) => void;
+  /**
+   * Checkbox to be shown before TreeNode.
+   * If undefined checkbox will not be shown.
+   * Recommended to use Checkbox component.
+   */
+  nodeCheckbox?: React.ReactNode;
+  /**
+   * Content shown after TreeNode.
+   */
+  children?: React.ReactNode;
+} & CommonProps;
+
+/**
+ * TreeNode component to display within a Tree.
+ * @example
+  <TreeNode
+    nodeId={props.nodeId}
+    label={props.node.label}
+    sublabel={props.node.subLabel}
+    onNodeExpanded={onNodeExpanded}
+    onNodeSelected={onSelectedNodeChange}
+    isDisabled={props.isDisabled}
+    isExpanded={props.isExpanded}
+    isSelected={props.isSelected}
+    nodeCheckbox={
+      <Checkbox variant='eyeball' disabled={props.isDisabled} />
+    }
+    icon={<SvgPlaceholder />}
+  />
+ */
+export const TreeNode = (props: TreeNodeProps) => {
+  const {
+    nodeId,
+    label,
+    sublabel,
+    children,
+    style,
+    className,
+    icon,
+    isDisabled = false,
+    isExpanded = false,
+    isSelected = false,
+    onNodeSelected,
+    onNodeExpanded,
+    nodeCheckbox,
+    ...rest
+  } = props;
+  useTheme();
+
+  const context = React.useContext(TreeContext);
+  const nodeDepth = context?.nodeDepth ?? 0;
+  const subNodeIds = context?.subNodeIds ?? [];
+
+  const styleLevel = React.useMemo(
+    () =>
+      getWindow()?.CSS?.supports?.(`--level: ${nodeDepth}`)
+        ? { '--level': nodeDepth, ...style }
+        : { marginLeft: nodeDepth ? nodeDepth * 28 : 0, ...style },
+    [nodeDepth, style],
+  );
+
+  const onNodeClick = () => {
+    if (isDisabled) {
+      return;
+    }
+    onNodeSelected?.(nodeId, !isSelected);
+  };
+
+  return (
+    <>
+      <li role='treeitem' id={nodeId} aria-expanded={isExpanded} {...rest}>
+        {
+          <div
+            className={cx(
+              'iui-tree-node',
+              {
+                'iui-active': isSelected,
+                'iui-disabled': isDisabled,
+              },
+              className,
+            )}
+            style={styleLevel}
+            onClick={() => onNodeClick()}
+            tabIndex={isSelected ? 0 : -1}
+          >
+            {nodeCheckbox && React.isValidElement(nodeCheckbox)
+              ? React.cloneElement(nodeCheckbox, {
+                  className: cx(
+                    'iui-tree-node-checkbox',
+                    nodeCheckbox.props.className,
+                  ),
+                })
+              : nodeCheckbox}
```

Comment:
What is the case, when `nodeCheckbox` is not a valid element, but we still want to display it?

Reply:
I guess there is no case for that, changed to `undefined` instead

Reply:
I think it is not a valid element when text is passed.

Reply:
So should I change it back so it will show text if `nodeCheckbox` is not a valid element?

Reply:
I dont think passing text could be taken as node checkbox... Opinions?

Reply:
Should we care what user passes?

Reply:
amm.. I guess yes, if we want Tree to work as expected :D or we can render that string I guess if nothing will break

Reply:
Ok I reverted it to `nodeCheckbox` so it will render if text was passed in